python/kserve/kserve/model_repository.py

```diff
@@ -30,7 +31,7 @@ class ModelRepository:
     """
 
     def __init__(self, models_dir: str = MODEL_MOUNT_DIRS):
-        self.models: Dict[str, Union[Model, DeploymentHandle]] = {}
+        self.models: Dict[str, Union[Model, OpenAIModel, DeploymentHandle]] = {}
```

Comment:
I think ideally we would have a base abstract class that all models inherit from. This may be a much larger change so adding `OpenAIModel` to the `Unions` works for now.

Reply:
Agree. Happy to add the base class in a follow up PR.

Reply:
I eventually had to convert the implementation to use a base class. Otherwise openai model was exposed to other components. This then required them to pull the openai dependency. I believe we don't want to impose openai support anywhere else.